src/main/java/htsjdk/tribble/gff/Gff3Writer.java

```diff
@@ -0,0 +1,103 @@
+package htsjdk.tribble.gff;
+
+import htsjdk.samtools.util.FileExtensions;
+import htsjdk.samtools.util.IOUtil;
+import htsjdk.tribble.TribbleException;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.zip.GZIPOutputStream;
+
+
+public class Gff3Writer implements Closeable {
+
+    private final PrintStream out;
```

Comment:
I would not use a PrintStream.  It doesn't throw exceptions when things go wrong during writing but instead sets a flag you have to manually check.  It's designed for writing to interactive consoles and should basically never be used when writing files.  It is used badly in  a number of places in htsjdk but we should avoid new ones and work on stamping out the old ones..  Maybe just a BufferedOutputStream and write linebreaks manually?   Does gff3 define a specific newline character or is it platform specific?

Reply:
Good to know, changes to BufferedOutputStream.